cpp/src/arrow/compute/exec/test_nodes.cc

```diff
@@ -60,7 +60,7 @@ class ConcatNode : public ExecNode {
         pause_if_above_(pause_if_above),
         resume_if_below_(resume_if_below) {
     for (std::size_t i = 0; i < inputs_.size(); i++) {
-      input_ptrs_into_queue_.push_back(kNone);
+      input_ptrs_into_queue_.push_back(std::nullopt);
```

Comment:
Why do we need to change this file?

Reply:
I believe this is [explained by Weston here](https://github.com/westonpace/arrow/pull/27#issuecomment-1346943850).

Reply:
But why is this an issue with this PR? Does your code change affect this file?

Reply:
It seems to me to be an orthogonal issue but I feel I am missing sth here

Reply:
It's orthogonal to this change, yes.  The Windows tests were broken I think before this change (I think this failure was masked by a DLL export error).  I have now finally got this test working on Windows so it might be nice to include but, given this is a unit test, probably not super important.

Reply:
kk sounds good